test/com/google/enterprise/adaptor/fs/WindowsFileDelegateTest.java

```diff
@@ -269,33 +218,26 @@ public void testGetDfsShareAclView() throws Exception {
     assertEquals(expectedAcl.getAcl(), aclView.getAcl());
   }
 
-  @Test
-  public void testGetDfsShareAclViewUnsupportedAceType() throws Exception {
-    thrown.expect(IllegalArgumentException.class);
-    AclFileAttributeView aclView = getDfsShareAclView(
-        new AceBuilder()
-        .setSid(AccountSid.group("SystemAuditAce", null))
-        .setType(WinNT.SYSTEM_AUDIT_ACE_TYPE)
-        .setPerms(WinNT.GENERIC_ALL)
-        .setFlags(WinNT.OBJECT_INHERIT_ACE, WinNT.CONTAINER_INHERIT_ACE)
-        .build());
-  }
-
-  private static AclFileAttributeView getDfsShareAclView(
+  private static AclFileAttributeView getDfsShareAclView(String path,
+      final int lastError, final boolean isLinkAcl,
       WinNT.ACCESS_ACEStructure... aces) throws Exception {
     final byte[] dacl = buildDaclMemory(aces);
+
     Kernel32Ex kernel32 = new UnsupportedKernel32() {
         @Override
         public int GetLastError() {
           // For when GetFileSecurity returns false.
-          return W32Errors.ERROR_INSUFFICIENT_BUFFER;
+          return lastError;
         }
       };
     Advapi32 advapi32 = new UnsupportedAdvapi32() {
         @Override
         public boolean GetFileSecurity(WString lpFileName,
             int RequestedInformation, Pointer pointer, int nLength,
             IntByReference lpnLengthNeeded) {
+          if (isLinkAcl || lastError != W32Errors.ERROR_INSUFFICIENT_BUFFER) {
```

Comment:
The DsfInfo3 change (below) prompted using a single mock for all these tests.  Previously, the individual mocks would support either returning significant data in the DsfInfo150 or GetFileSecurity, but not both.  For a DFS Namespace, only GetFileSecurity will be called.  For a DFS Link, NetDfsGetInfo(level 150) will be called, if it returns null, then GetFileSecurity will be called.